cni-plugin/internal/pkg/utils/utils.go

```diff
@@ -320,29 +322,40 @@ func ReplaceHostLocalIPAMPodCIDRs(logger *logrus.Entry, stdinData map[string]int
 	return nil
 }
 
-func replaceHostLocalIPAMPodCIDR(logger *logrus.Entry, rawIpamData interface{}, getPodCidr func() (string, error)) error {
+func replaceHostLocalIPAMPodCIDR(logger *logrus.Entry, rawIpamData interface{}, getPodCidrs func() (string, string, error)) error {
 	logrus.WithField("ipamData", rawIpamData).Debug("Examining IPAM data for usePodCidr")
 	ipamData, ok := rawIpamData.(map[string]interface{})
 	if !ok {
 		return fmt.Errorf("failed to parse host-local IPAM data; was expecting a dict, not: %v", rawIpamData)
 	}
 	subnet, _ := ipamData["subnet"].(string)
+	ipv4Cidr, ipv6Cidr, err := getPodCidrs()
```

Comment:
I believe this is the change that is causing the majority of the tests to fail. Note that the tests in question don't actually use `usePodCidr` in their config: https://github.com/projectcalico/calico/blob/master/cni-plugin/tests/calico_cni_k8s_test.go#L197-L200

So, we need to be capable of running without querying from the k8s node, and using just the given CIDR.

If you look at the original code, we only queried the k8s API when `usePodCidr` was present, but the new code does this unconditionally.

I think we need something like this here now:


```go
var ipv4Cidr, ipv6Cidr string
var err error
if strings.EqualFold(subnet, "usePodCidr") || strings.EqualFold(subnet, "usePodCidrIPv6") {
        ipv4Cidr, ipv6Cidr, err = getPodCidrs()
        if err != nil {
                logger.Errorf("Failed to getPodCidrs")
                return err
        }
}
```

Reply:
That makes total sense. Thanks a lot Casey, I was in a rabbithole :)